test/models/filter_test.rb

```diff
@@ -13,13 +12,16 @@ class FilterTest < ActiveSupport::TestCase
   should have_many(:event_type_filters)
   should have_many(:event_types).through(:event_type_filters)
 
+  should belong_to(:account)
+
   before do
     Builder::App.build! do |a|
       a.name 'not_useful'
       event_types.each { |type| a.event_type type }
     end
   end
 
+  let(:account) { FactoryBot.create(:account) }
```

Comment:
Filters should also be scoped by account - it means the SQL query should be affected by this assumption.